### PR TITLE
Merging dgi code analyzer with minerva java analyzer and improvements done for sdg generation

### PR DESCRIPTION
**Pull request by fabio-franco**
As part of this task, there are 5 new messages created:

```
CallGraphFileIsNotJAR = The file {0} is not a JAR and it will be skipped.
CallGraphAddExtraLibToScope = Adding dependency library to analysis scope: {0}
CallGraphBuildMethodLevel = Building the call graph with method level information ...
CallGraphPopulatingMethodLevel = Populating call graph with method level information ...
CallGraphCalculatingWeightMethodLevel = Calculating the weight for the connections in the call graph with method level information ...
```

### CODE REVIEW

File: src/main/java/com/ibm/minerva/analyzer/Analyzer.java [hunk 1]
```diff
@@ -95,9 +111,42 @@ public Analyzer setCallGraphBuilder(CallGraphBuilderType type) throws IOExceptio
...
+        		else {
+	            	addExtraLibrary(extraLib);
+	            }
+			}
+        }
+    }
 
     public void run() throws IOException {
         try {
+        	long start = System.currentTimeMillis();
```
Comment (fabio-franco) [inline src/main/java/com/ibm/minerva/analyzer/Analyzer.java]: need to be removed

File: src/main/java/com/ibm/minerva/analyzer/Analyzer.java [hunk 2]
```diff
@@ -118,7 +167,16 @@ public void run() throws IOException {
...
+                
+                // Add extra libraries, if is there any
+                if (additionalLibraries != null && additionalLibraries.size() > 0) {
+                	archiveProcessor.processExtraLibs(additionalLibraries.toArray(new File[additionalLibraries.size()]));
+                }
             }
+            
+            long end = System.currentTimeMillis();
+            
+            logger.info("Time (in milliseconds) to process all classes: "+(end-start));
```
Comment (fabio-franco) [inline src/main/java/com/ibm/minerva/analyzer/Analyzer.java]: need to be removed

File: src/main/resources/com/ibm/minerva/analyzer/messages.properties [hunk 3]
```diff
@@ -51,3 +51,9 @@ CallGraphTempFileWriteWarning = Class {0} will be processed in memory because it
 CallGraphTempFileReadError = An error occurred while attempting to read the temporary file for class {0}. Message: {1}
 CallGraphBuildError = An unexpected error occurred while attempting to build the call graph. Message: {0}
 CallGraphWriteError = An unexpected error occurred while attempting to write the call graph to file {0}. Message: {1}
+
+CallGraphFileIsNotJAR = The file {0} is not a JAR and it will be skipped.
```
Comment (lentheivendra) [inline src/main/resources/com/ibm/minerva/analyzer/messages.properties]: Just need to drop the "it"

File: src/main/java/com/ibm/minerva/analyzer/Analyzer.java (hunk 1 above)
Comment (lentheivendra) [inline src/main/java/com/ibm/minerva/analyzer/Analyzer.java]: As discussed, can remove all temporary time captures etc

**Review by lentheivendra — APPROVED** [review]
LGTM

**Comment by fabio-franco** [comment]
During a clean up I removed the verifications inside the while loop in SDGGraph2JSON.java, but they are important to avoid a null pointer exception in jpet, so I am returning back those extra verifications.

**Pull request merged**